### PR TITLE
Add script to generate pin control headers and pin control groups for iMX RT parts

### PR DESCRIPTION
**Pull request by danieldegrasse**
Add initial script to generate pinctrl groups from MEX file for iMX.RT
parts. This script uses the files provided in MCUXpresso config tools offline data
(downloadable from MCUXpresso SDK builder), combined with a MEX configuration file (editable with MCUXpresso config tools),
to automatically generate pinctrl groups for a given board.

Pinctrl groups can be handcoded without the aid of this script, it is only made available to simplify generation of pinctrl groups for NXP boards.

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@mmahadevan108, yes, and the board dtsi file included in the [Zephyr PR](https://github.com/zephyrproject-rtos/zephyr/pull/41810) is also generated by this script

**Review by dleach02 — APPROVED** [review]

File: mcux/scripts/rt_cfg_utils.py [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+        # Build dictionary mapping pin properties to pins. This allows us to
+        # group pins based on shared configuration
+        self._pin_groups = collections.defaultdict(lambda: [])
+        for pin in pins:
+            signal_name = pin.attrib.get('pin_signal')
+            signal = signal_map[signal_name]
+            if not signal:
+                print(f"Warning: Signal name {signal_name} not present in mapping")
+                # No way to find mux option
+                continue
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply mcux/scripts/rt_cfg_utils.py]: Added this change in the latest push. Out of interest, what signal was causing a `KeyError`? Some signals require no configuration by the pinctrl driver, and so aren't placed in the signal mapping. I want to make sure this is one of those signals and that the pinctrl driver isn't missing the ability to configure some pins.

**Comment by danieldegrasse** [comment]
@dleach02 This PR should be good to merge now that the pin control PR in Zephyr has gone in.

**Comment by NickolasLapp** [comment]
Two other things that I noticed on this PR, for example when trying to generate pinctrl for the RT1060-EVKB. 
Command:
```
python3 rt_gen_dts.py --cfg-tool-root ~/Downloads/MIMXRT1060-EVKB_ConfigTools_data/processors /Users/nickolas@whisper.ai/Downloads/MIMXRT1060-EVKB_ConfigTools_data/boards/MIMXRT1060-EVKB/ksdk2_0/MIMXRT1060-EVKB.mex
file
```

1. Had to change namespaces to: `NAMESPACES = {'mex' : 'http://mcuxpresso.nxp.com/XSD/mex_configuration_1.9'}`
* The default (`mex_configuration_10`) didn't work for the mex file I downloaded for the EVKB. Not sure what the proper solution is here
2. I had to manually copy over the signal configuration reference.
* Inside the package `MIMXRT1062DVL6A/signal_configuration.xml`, rather than specifying the full signal configuration, it simply provides a reference: `<reference file="MIMXRT1062CVJ5A/signal_configuration.xml"/>`. When I ran the script by default, I would get an error that there were no pins specified in the signal map. I was able to resolve the issue by manually "following" the reference, rather copying the reference files pin specs over to the first file, and then the .dtsi was able to generate.

File: mcux/scripts/rt_cfg_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply mcux/scripts/rt_cfg_utils.py]: Those do all look like special function pins to me. I wouldn't worry about the warnings.

**Comment by danieldegrasse** [comment]
> Had to change namespaces to: NAMESPACES = {'mex' : 'http://mcuxpresso.nxp.com/XSD/mex_configuration_1.9'}

@NickolasLapp this is interesting. That namespace versioning number looks off (I'd expect that value to go up, not down). When I attempt a generation using the default MEX file for the EVKB, I don't have that issue (I've attached the file I used, as a ZIP). I created this file using the new file dialogue in MCUXpresso config tools, then saved the file that was generated by the tool.

> I had to manually copy over the signal configuration reference

This is an issue I've seen, but in the data packs I have downloaded directly from the MCUXpresso SDK generator I haven't seen this structure being used. Parsing these references is be a feature I may add to this script, since following a reference like that is relatively easy to encode.

[MIMXRT1060-EVKB.mex.zip](https://github.com/zephyrproject-rtos/hal_nxp/files/8374670/MIMXRT1060-EVKB.mex.zip)

**Comment by danieldegrasse** [comment]
@NickolasLapp I've updated this script to use the v11 configuration files from MCUXpresso config tools. Could you verify that the tool still works for you?

**Pull request merged**